core/src/main/scala/influxdbreporter/core/Reportable.scala

```diff
@@ -88,31 +71,53 @@ trait Reschedulable {
   def reschedule(): Unit
 }
 
-private class StoppableReportingTaskWithRescheduling(scheduler: ScheduledExecutorService,
-                                                     createTask: Reschedulable => Runnable,
-                                                     delay: FiniteDuration,
-                                                     client: MetricClient[_])
-  extends StoppableReportingTask with Reschedulable with LazyLogging {
+private trait MetricsReporter {
+  def reportMetrics(): Future[Unit]
+
+  def stop(): Unit
+}
+
+private class StoppableReportingTaskWithRescheduling(metricsReporter: MetricsReporter,
```

Comment:
Compiler's "safe init" option seems more strict for Scala 3. This file and those classes needed some refactoring in order to avoid loops when initializing them. Behavior should be the same and the change was designed as binary compatible.